### PR TITLE
Move CDXJ file handling from bin/cli.js to WACZ class

### PR DESCRIPTION
**Pull request by tw4l**
Follow-up PR for issue #88 

This change makes it so that other applications like Browsertrix Crawler that want to use js-wacz as a library can pass in existing CDXJ files simply by setting the right values during the WACZ class initialization rather than having to duplicate code from the js-wacz CLI module to add each line of the provided CDXJ files to the right b-tree manually.

### CODE REVIEW

**Review by matteocargnelutti — APPROVED** [review]
Hey there @tw4l ! Great PR, thank you. 

I added docs for this new `cdjx` option in `types.js` and made a few other related edits - would you mind having a look?  

Thank you 👋

File: types.js [hunk 1]
```diff
@@ -3,16 +3,17 @@
...
+ * @property {boolean} [detectPages=true] - If true (default), will attempt to detect pages in WARC records to generate a pages.jsonl file. Automatically disabled if: `pages`/ `cdxj` is provided, or `addPages()`/ `addCDJX()` is called.
  * @property {?string} url - If set, will be added to datapackage.json as `mainPageUrl`.
  * @property {?string} ts - If set, will be added to datapackage.json as `mainPageDate`. Can be any value that `Date()` can parse.
  * @property {?string} title - If set, will be added to datapackage.json as `title`.
  * @property {?string} description - If set, will be added to datapackage.json as `description`.
  * @property {?string} signingUrl - If set, will be used to try and sign the resulting archive.
  * @property {?string} signingToken - Access token to be used in combination with `signingUrl`.
  * @property {?Object} datapackageExtras - If set, will be appended to datapackage.json under `extras`.
+ * @property {?string} cdxj - If set, skips indexing and allows for passing CDXJ files "as is". Path to a folder containing CDXJ files. Allows
```
Comment (tw4l) [inline types.js:15]: Thanks for adding these! Looks like maybe some text got cut off at the end here?=

Comment (matteocargnelutti) [reply types.js]: Woops, thanks!

**Comment by matteocargnelutti** [comment]
@tw4l Merging now to make upcoming work on #119 easier :) thanks again!

**Pull request merged**